lib/buffer.c

```diff
@@ -25,8 +25,8 @@ size_t fuse_buf_size(const struct fuse_bufvec *bufv)
 	size_t size = 0;
 
 	for (i = 0; i < bufv->count; i++) {
-		if (bufv->buf[i].size == SIZE_MAX)
-			size = SIZE_MAX;
+		if (bufv->buf[i].size >= SIZE_MAX - size)
+			return SIZE_MAX;
 		else
```

Comment:
Thanks, objections to remove this else? I don't think we need it?

Reply:
@bsbernd  Yes, the else is not needed, done as your advice!